graveler/committed/meta_range_writer.go

```diff
@@ -156,11 +158,16 @@ func (w *GeneralMetaRangeWriter) shouldBreakAtKey(key graveler.Key) bool {
 
 // writeRangesToMetaRange writes all ranges to a MetaRange and returns the MetaRangeID
 func (w *GeneralMetaRangeWriter) writeRangesToMetaRange() (*graveler.MetaRangeID, error) {
-	metaRangeWriter, err := w.metaRangeManager.GetWriter(w.ctx, w.namespace)
+	metaRangeWriter, err := w.metaRangeManager.GetWriter(w.ctx, w.namespace, w.metadata)
 	if err != nil {
 		return nil, fmt.Errorf("failed creating metarange writer: %w", err)
 	}
 	metaRangeWriter.AddMetadata(MetadataTypeKey, MetadataMetarangesType)
+	// write user provided metadata, if any
+	for k, v := range w.metadata {
+		metaRangeWriter.AddMetadata(k, v)
+	}
```

Comment:
1. Set the metadata before setting the type so it will not override the built-in metadata we always want to set
2. General note we need to change AddMetadata to SetMetadata it overrides - or just expose the Map as http's request Header()

Reply:
Done